### ISSUE TITLE
[Master feature] True doc-level infinite scroll

### ISSUE THREAD

**Issue by wassgha** [ampproject/amphtml#25500]
Master issue to track work on fixing up `amp-next-page` and adding true infinite scroll to it. Will be updated with more tasks as needed.

Replaces #14059

#### Fixing broken features
- [x] `amp-pixel` preemptively firing event on fetch #25069 
- [x] Broken behavior on smaller documents #14876 
- [x] Broken transitions to the first page on large viewports #19823 
- [x] Lack of `amp-analytics` support (#15807 , #18735 , and #20400 )
- [x] Large bundle size because of dependence on `v0` #22357
- [ ] Interaction between the recommendation box and the AMP cache #15700
#### Adding new features
- [ ] End-to-end tests #24610
- [x] Start v0.2 implementation
- [x] Dynamic adoption of shadow mode #25630
- [ ] Communicate document changes to the viewer #15819
- [x] Better separators and transitions (TBD, #15818 is partial)
- [x] Better handling of sticky elements (#26106 )
- [x] True infinite scroll phase 1
- [x] True infinite scroll phase 2 (unloading, placeholding and reloading, #26296)
- [x] Templating the recommendation box (#26610 )
- [x] Allow for history manipulation (minimally to be able to go back to the original article) #25553

**Comment by nainar** [comment]
@wassgha can this be closed?

**Comment by wassgha** [comment]
Yup!
